tensorflow_cloud/deploy.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import random
+import string
+import subprocess
+
+from googleapiclient import discovery
+from googleapiclient import errors
+
+from . import gcp
+
+
+def deploy_job(region, image_uri, chief_config, worker_count, worker_config,
+               entry_point_args, enable_stream_logs):
+    job_name = _get_name()
+    project_id = gcp.get_project_name()
+    ml_apis = discovery.build('ml', 'v1')
+
+    request_dict = _create_request_dict(
+        job_name, region, image_uri, chief_config, worker_count, worker_config,
+        entry_point_args)
+    try:
+        response = ml_apis.projects().jobs().create(
+            parent='projects/{}'.format(project_id),
+            body=request_dict
+        ).execute()
+        print('Job submitted successfully.')
+        _print_logs_info(job_name, project_id)
+        # TODO(psv): Add support for streaming logs.
+    except errors.HttpError as err:
+        print('There was an error submitting the job.')
```

Comment:
Prefer re-raising rather than printing (so that the caller of the code can `except`)

Reply:
Raising a RuntimeError with the error reason.